### PR TITLE
feat(chart): Add expression, description and verbose name to search filter

### PR DESCRIPTION
**Pull request by nikolagigic**
### SUMMARY
Explore 2021 Q1 roadmap item https://github.com/apache-superset/superset-roadmap/issues/141
Add **expression**, **description** and **verbose_name** to the search functionality for **metrics** and **columns** in chart **datasource panel** and sort by relevance.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screenshot 2021-01-15 at 15 03 43

### TEST PLAN

1. Go to any **chart**
2. In the **datasource panel** hover over any metric/column **question mark/info icon** and note the text inside if any
3. Search for any portion of the noted text

Search should now filter columns/metrics based on those fields as well.

### ADDITIONAL INFORMATION

- [ ] Has associated issue: roadmap item https://github.com/apache-superset/superset-roadmap/issues/141
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by villebro — APPROVED** [review]
Code changes LGTM and I agree with adding these to search scope

**Comment by junlincc** [comment]
@nikolagigic thanks Nikola! do you plan to address the result ranking request in a separate PR?

**Review by junlincc — APPROVED** [review]
LGTM, Awesome! Data panel is gradually reaching its full potential. thank you @nikolagigic 
cc @pkdotson @ktmud @graceguo-supercat 

Screen Shot 2021-01-18 at 1 44 19 AM
Screen Shot 2021-01-18 at 1 32 31 AM

**Review by zhaoyongjie — APPROVED** [review]
The code LGTM. 

image

**Comment by junlincc** [comment]
@nikolagigic let's open another PR to take care of the search result ranking, and get this one in first.

File: superset-frontend/src/explore/components/DatasourcePanel.tsx [hunk 1]
```diff
@@ -175,8 +175,12 @@ const DataSourcePanel = ({
       return;
     }
     setList({
-      columns: matchSorter(columns, value, { keys: ['column_name'] }),
-      metrics: matchSorter(metrics, value, { keys: ['metric_name'] }),
+      columns: matchSorter(columns, value, {
+        keys: ['column_name', 'expression', 'description', 'verbose_name'],
+      }),
+      metrics: matchSorter(metrics, value, {
+        keys: ['metric_name', 'expression', 'description', 'verbose_name'],
```
Comment (ktmud) [inline superset-frontend/src/explore/components/DatasourcePanel.tsx:182]: description and verbose_name should probably rank before expression.

Comment (nikolagigic) [reply superset-frontend/src/explore/components/DatasourcePanel.tsx:182]: got it

**Pull request merged**